client/my-sites/stats/jetpack-upsell-section/index.tsx

```diff
@@ -25,13 +26,14 @@ const QUERY_VALUES = {
 
 export default function JetpackUpsellSection() {
 	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state ) );
+	const hasPaidStats = useSelector( hasSiteProductJetpackStatsPaid );
 
 	// NOTE: This will only work within Odyssey Stats.
-	const { purchasedProducts, error, isLoading } = usePurchasedProducts();
+	const { purchasedProducts } = usePurchasedProducts();
 
 	// Exit early if we don't have and can't get the site purchase data.
 	// Also exit early if we're not in the Odyssey Stats environment.
-	if ( isLoading || error || ! isOdysseyStats ) {
```

Comment:
Are we now handling the loading/error case elsewhere?

Reply:
We are using the purchase array in the initial state, where worse case scenario, the product list would be empty. 😄